### PR TITLE
Fix stannp in test mode

### PR DESCRIPTION
**Pull request by richtier**
[fix this woe](https://sentry.ci.uktrade.io/dit/directory-api-dev/issues/1832/)

### CODE REVIEW

File: company/stannp.py [hunk 1]
```diff
@@ -12,21 +12,7 @@ def __init__(self, api_key, test_mode=True):
         self.test_mode = test_mode
 
     def post(self, url, data):
-        response = requests.post(
-            url, data=data, auth=(self.api_key, '')
-        )
-        return response.json()
-
-    def validate_recipient(self, recipient):
```
Comment (richtier) [inline company/stannp.py]: is not used anywhere

Comment (konradko) [inline company/stannp.py]: I guess the validation happens when you POST the letter as well, so cool

File: company/tests/test_stannp.py [hunk 2]
```diff
@@ -1,50 +1,68 @@
...
-                'country': 'test_country',
-                'custom_fields': [
-                    ('test_field_name1', 'test_value1'),
-                    ('test_field_name2', 'test_value2'),
-                ]
-            },
-        )
-
-    mock_requests.assert_called_once_with(
+    assert mock_post.call_count == 1
```
Comment (konradko) [inline company/tests/test_stannp.py]: why replace `assert_called_once_with`?

Comment (richtier) [inline company/tests/test_stannp.py]: mocks will not comain if this happens:

`assert_called_once_wit(`
`asert_called_once_with(`

etc

https://engineeringblog.yelp.com/2015/02/assert_called_once-threat-or-menace.html

**Pull request merged**